File: glowxq-api/business/business-oj/src/main/java/com/glowxq/oj/topic/enums/OiScoreType.java
package com.glowxq.oj.topic.enums;

import com.glowxq.core.common.enums.base.BaseType;
import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * 得分类型
 *
 * @author glowxq
 * @version 1.0
 * @date 2025/1/20
 */
@AllArgsConstructor
@Getter
public enum OiScoreType implements BaseType {

    // 最近得分
    Recent("Recent", "最近得分"),

    // 最高得分
    Highest("Highest", "最高得分");

    /**
     * code
     */
    private final String code;

    /**
     * 最近类型
     */
    private final String name;

    /**
     * 根据code获取枚举
     *
     * @param code
     * @return
     */
    public static OiScoreType matchCode(String code) {
        for (OiScoreType pushStatus : OiScoreType.values()) {
            if (pushStatus.getCode().equals(code)) {
                return pushStatus;
            }
        }
        return null;
    }
}

File: oj-web/src/enums/oj/topic/OiScoreType.ts
import type { BaseEnum } from '@/enums/base';

// OI 得分类型枚举接口
interface OiScoreTypeEnum extends BaseEnum {
  tooltip?: string;
  text?: string;
}

/**
 * OI 得分类型枚举
 * 用于标识不同的得分类型
 */
export const OiScoreType: { [key: string]: OiScoreTypeEnum } = {
  /**
   * 最近得分
   */
  RECENT: {
    code: 'Recent',
    name: '最近得分',
    tooltip: '使用最近一次提交的得分',
    text: '使用最近一次提交的得分'
  },

  /**
   * 最高得分
   */
  HIGHEST: {
    code: 'Highest',
    name: '最高得分',
    tooltip: '使用历史提交中的最高得分',
    text: '使用历史提交中的最高得分'
  }
};

/**
 * 根据code获取枚举
 * @param code
 * @returns 枚举值或null
 */
export function matchCode(code: string): OiScoreTypeEnum | null {
  return Object.values(OiScoreType).find(item => item.code === code) || null;
} 